src/providers/authentication/eIDProvider.ts

```diff
@@ -0,0 +1,116 @@
+import {Dispatch, SetStateAction} from 'react';
+import {addMessageListener, AusweisAuthFlow, AusweisSdkMessage, sendCommand} from '@animo-id/expo-ausweis-sdk';
+import {PARMode} from '@sphereon/oid4vci-common';
+import {OpenID4VCIClient} from '@sphereon/oid4vci-client';
+import {EIDFlowState, EIDGetAuthorizationCodeArgs, EIDHandleErrorArgs, EIDInitializeArgs, EIDProviderArgs} from '../../types';
+
+class EIDProvider {
+  private readonly client: OpenID4VCIClient;
+  private readonly tcTokenUrl: string;
+  private readonly onStateChange?: Dispatch<SetStateAction<EIDFlowState>> | ((status: EIDFlowState) => void);
+  private authFlow: AusweisAuthFlow;
+  public currentState: EIDFlowState;
+
+  private constructor(args: EIDProviderArgs) {
+    const {client, onStateChange, onAuthenticated, onEnterPin, tcTokenUrl} = args;
+
+    this.tcTokenUrl = tcTokenUrl;
+    this.client = client;
+    this.onStateChange = onStateChange;
+    this.authFlow = new AusweisAuthFlow({
+      onEnterPin,
+      onError: (error): void => {
+        this.handleError(error);
+      },
+      onSuccess: (options): void => {
+        this.getAuthorizationToken(options).then((authorizationCode: string) => onAuthenticated(authorizationCode));
+      },
+      onInsertCard: (): void => {
+        this.handleStateChange({state: 'INSERT_CARD'});
+      },
+    });
+
+    this.handleStateChange({state: 'INITIALIZED'});
+  }
+
+  public static async initialize(args: EIDInitializeArgs): Promise<EIDProvider> {
+    const uri: string =
+      'openid-credential-offer://?credential_offer=%7B%22credential_issuer%22%3A%22https%3A%2F%2Fdemo.pid-issuer.bundesdruckerei.de%2Fc%22%2C%22credential_configuration_ids%22%3A%5B%22pid-sd-jwt%22%5D%2C%22grants%22%3A%7B%22authorization_code%22%3A%7B%7D%7D%7D';
+    const client = await OpenID4VCIClient.fromURI({
+      uri,
+      resolveOfferUri: true,
+      retrieveServerMetadata: true,
+      clientId: 'bc11dd24-cbe9-4f13-890b-967e5f900222',
+      // This is a separate call, so we don't fetch it here, however it may be easier to just construct it here?
+      createAuthorizationRequestURL: false,
+    });
+
+    const authorizationRequestUrl = await client.createAuthorizationRequestUrl({
+      authorizationRequest: {
+        redirectUri: 'https://sphereon.com/redirect',
+        scope: 'pid', // TODO we need to determine the scopes using the metadata
```

Comment:
No, since this Provider is tied to the German PID Issuer, we actually better have a fixed scope

Reply:
removed the comment